ssh-pubkey/ssh-pubkey

```diff
@@ -2,43 +2,99 @@
+    echo '    #           and update your ~/.ssh/config accordingly'
+    echo "    mv ~/.ssh/id_rsa ~/.ssh/id_rsa.${my_ts}.bak"
+    # shellcheck disable=SC2016
+    echo '    ssh-keygen -t ed25519 -f ~/.ssh/id_ed25519 -q -N ""'
+    echo "    echo 'Host *
+        IdentityFile ~/.ssh/id_ed25519
+        IdentityFile ~/.ssh/id_rsa.${my_ts}.bak' >> ~/.ssh/config"
+
+    echo ''
+    echo '    # OPTION 2: Generate a larger, 4096-bit RSA key"'
+    echo '    #           and update your ~/.ssh/config accordingly'
+    echo "    mv ~/.ssh/id_rsa ~/.ssh/id_rsa.${my_ts}.bak"
+    # shellcheck disable=SC2016
+    echo '    ssh-keygen -t rsa -b 4096 -f ~/.ssh/id_rsa -q -N ""'
+    echo "    echo 'Host *
+        IdentityFile ~/.ssh/id_rsa
+        IdentityFile ~/.ssh/id_rsa.${my_ts}.bak' >> ~/.ssh/config"
+); }
+
 main() {
 
-    if [ ! -d "$HOME/.ssh" ]; then
-        mkdir -p "$HOME/.ssh/"
-        chmod 0700 "$HOME/.ssh/"
+    if ! test -d ~/.ssh; then
+        mkdir -p ~/.ssh/
+        chmod 0700 ~/.ssh/
     fi
 
-    if [ ! -f "$HOME/.ssh/config" ]; then
+    if ! test -f ~/.ssh/config; then
         # for the benefit of VSCode
-        touch "$HOME/.ssh/config"
-        chmod 0644 "$HOME/.ssh/config"
+        touch ~/.ssh/config
+        chmod 0644 ~/.ssh/config
     fi
 
-    if [ ! -f "$HOME/.ssh/authorized_keys" ]; then
-        touch "$HOME/.ssh/authorized_keys"
-        chmod 0600 "$HOME/.ssh/authorized_keys"
+    if ! test -f ~/.ssh/authorized_keys; then
+        touch ~/.ssh/authorized_keys
+        chmod 0600 ~/.ssh/authorized_keys
     fi
 
-    if [ ! -f "$HOME/.ssh/id_rsa" ]; then
-        ssh-keygen -b 2048 -t rsa -f "$HOME/.ssh/id_rsa" -q -N ""
-        echo >&2 ""
-    fi
-
-    if [ ! -f "$HOME/.ssh/id_rsa.pub" ]; then
-        ssh-keygen -y -f "$HOME/.ssh/id_rsa" > "$HOME/.ssh/id_rsa.pub"
+    my_keytype=''
+    if test -f ~/.ssh/id_ed25519; then
+        my_keytype='ed25519'
+        if ! test -f ~/.ssh/id_ed25519.pub; then
+            echo >&2 ""
+            ssh-keygen -y -f ~/.ssh/id_ed25519 > ~/.ssh/id_ed25519.pub
+        fi
+    elif test -f ~/.ssh/id_rsa; then
+        my_keytype='rsa'
+        if ! test -f ~/.ssh/id_rsa.pub; then
+            echo >&2 ""
+            ssh-keygen -y -f ~/.ssh/id_rsa > ~/.ssh/id_rsa.pub
+        fi
+    elif test -f ~/.ssh/id_ecda; then
+        my_keytype='ecdsa'
+        if ! test -f ~/.ssh/id_ecdsa.pub; then
+            echo >&2 ""
+            ssh-keygen -y -f ~/.ssh/id_ecda > ~/.ssh/id_ecda.pub
+        fi
+    else
+        my_keytype='rsa'
         echo >&2 ""
+        ssh-keygen -b 4096 -t rsa -f ~/.ssh/id_rsa -q -N ""
+        ssh-keygen -y -f ~/.ssh/id_rsa > ~/.ssh/id_rsa.pub
     fi
 
     # TODO use the comment (if any) for the name of the file
     echo >&2 ""
     #shellcheck disable=SC2088
-    echo >&2 "~/Downloads/id_rsa.$(whoami).pub":
+    echo >&2 "~/Downloads/id_${my_keytype}.$(whoami).pub":
     echo >&2 ""
-    rm -f "$HOME/Downloads/id_rsa.$(whoami).pub"
```

Comment:
like this is still `$HOME`

Reply:
It has to be quoted for the "$(whoami)". Options are:

1. `rm -f "$HOME/Downloads/id_rsa.$(whoami).pub"`
2. `rm -f ~/Downloads/id_rsa."$(whoami)".pub`
3. `rm -f ~/"Downloads/id_rsa.$(whoami).pub"`

Thoughts?